### ISSUE TITLE
Artifacts: add HTTP (RFC9110) concepts

### ISSUE THREAD

**Issue by ioggstream** [d3fend/d3fend-ontology#100]
## I Expect

- to align concepts with RFC9110 
- to add missing concepts from RFC9110 (e.g. HTTP Intermediary, including HTTP gateways, HTTP reverse proxies & co)

## Question

Feel free to let me know whether this is ok or not.

**Comment by netfl0** [comment]
This is a excellent idea, please go ahead with modeling this. Let me know if you’ve got questions on our approach in OWL2DL. We model as classes and restrictions for relationships.

**Comment by aamedina** [comment]
There is a W3C working group note for a HTTP vocabulary in RDF (https://www.w3.org/TR/HTTP-in-RDF10/) which is used by the Thing Description Model which has become a recommendation: https://www.w3.org/TR/wot-thing-description11/#http-binding-assertions

Might be interesting to align here with OCSF https://schema.ocsf.io/1.3.0-dev/classes/http_activity

**Comment by ioggstream** [comment]
@aamedina interesting.  it would be useful to review and ensure that https://www.w3.org/TR/HTTP-in-RDF10/ is consistent with RFC9110 though.

HTTP is about semantics and abstracts concepts from the layers beneath (e.g. https://datatracker.ietf.org/doc/html/rfc9110#connections)

**Comment by aamedina** [comment]
I agree--it would be really interesting to see the semantics modeled from RFC9110 as per your suggestion, my preference should you develop this further is to favor class alignment to the existing HTTP RDF vocabulary so that that "Things" (in the "Web of Things") can be modeled as digital artifacts as others will begin to use IRIs with the predicates from that vocabulary when specifying IoT affordances. If IoT platforms begin to support the Thing Description model (with interaction affordances being bound to HTTP actions), I would appreciate alignment to automatically apply d3fend based threat models to IoT architectures.
